get-parity.sh

```diff
@@ -39,103 +37,56 @@ get_package() {
-	fi
-
-	if [ "$PKG" = "darwin" ] ; then
-		MD=$(curl -Ss ${LOOKUP_URL} | grep -v sha256 | grep -v md5 | grep pkg )
-		DOWNLOAD_FILE=$(echo $MD | cut -d "(" -f2 | cut -d ")" -f1)
-	fi
+  MD=$(curl -Ss ${LOOKUP_URL} | grep -v sha256 | grep " \[parity\]")
+  DOWNLOAD_FILE=$(echo $MD | grep -oE 'https://[^)]+')
 }
 
 check_upgrade() {
 
-	if [ -f /usr/bin/parity ] ; then
-		OLD_VERSION=$(parity --version | grep version|  cut -d/ -f2  | cut -d- -f1 | sed 's/v//g')
-	else
-		OLD_VERSION="0.0.0"
-	fi
-
-	if [ "$PKG" = "linux" ] ; then
-		FILE=$(curl -Ss $LOOKUP_URL | grep amd | cut -d "(" -f2 | cut -d ")" -f1)
-		NEW_VERSION=$(basename $FILE | cut -d_ -f2)
-	fi
-
-	if [ "$PKG" = "debian" ] ; then
-		NEW_VERSION=$(basename $DOWNLOAD_FILE  |  cut -d_ -f2)
-	fi
-
-	if [ "$PKG" = "centos" ] ; then
-	    NEW_VERSION=$(basename $DOWNLOAD_FILE  |  cut -d_ -f2)
+  # Determin new Version 
+  case "$RELEASE" in
+    "beta")  NEW_VERSION=$VERSION_BETA
+        ;;
+    "stable") NEW_VERSION=$VERSION_STABLE
+        ;;
+    *) NEW_VERSION=$(echo $DOWNLOAD_FILE | grep -oE 'v[0-9]+\.[0-9]+\.[0-9]+' | tr -d 'v')   
+        ;;
+  esac
 
-	fi
+  # Determin old (installed) Version 
+  parity_bin=$(which parity)
 
-	if [ "$PKG" = "darwin" ] ; then
-		NEW_VERSION=$(basename $DOWNLOAD_FILE | cut -d- -f2)
+	if [ -z $parity_bin ] ; then
+		OLD_VERSION="0.0.0"
+	else
+		OLD_VERSION=$(parity --version | grep -oE 'v[0-9]+\.[0-9]+\.[0-9]+' | tr -d 'v')
 	fi
 
 	if [ "$NEW_VERSION" = "$OLD_VERSION" ] ; then
 		echo "Parity $NEW_VERSION already installed"
 		exit 1
 	fi
-	if  version_gt "$NEW_VERSION" "$OLD_VERSION"  ; then
+
+  if  version_gt "$NEW_VERSION" "$OLD_VERSION"  ; then
 		echo "Upgrading parity from $OLD_VERSION to $NEW_VERSION"
 	else
 		echo "Existing version of parity: $OLD_VERSION is newer than the version you attempting to install: $NEW_VERSION"
 		exit 1
 	fi
 }
 
-
 install() {
+  TMPDIR=$(mktemp -d) && cd $TMPDIR
+	curl -Ss -O $DOWNLOAD_FILE
+  check_sha256
 
-    TMPDIR=$(mktemp -d)
-    cd $TMPDIR
-	$(curl -Ss -O $DOWNLOAD_FILE)
-
-	if [ "$PKG" = "debian" ] ; then
-		NAME=$(basename $DOWNLOAD_FILE)
-		sudo dpkg -i $TMPDIR/$NAME
-	fi
-
-	if [ "$PKG" = "linux" -a $LIBSSL = "10" ] ; then
-		NAME=$(basename $DOWNLOAD_FILE)
-		sudo dpkg -i $TMPDIR/$NAME
-	fi
-
-	if [ "$PKG" = "linux" -a $LIBSSL = "undef" ] ; then
-	    sudo cp $TMPDIR/parity /usr/bin
-		sudo chmod +x /usr/bin/parity
-	fi
-
-        if [ "$PKG" = "centos" ] ; then
-	    NAME=$(basename $DOWNLOAD_FILE)
-	    sudo rpm -U $TMPDIR/$NAME
+	if [ "$PKG" = "linux" ] ; then
+	  sudo cp $TMPDIR/parity /usr/bin && sudo chmod +x /usr/bin/parity
 	fi
 
 	if [ "$PKG" = "darwin" ] ; then
-		NAME=$(basename $DOWNLOAD_FILE)
-		sudo /usr/sbin/installer -pkg $TMPDIR/$NAME -target / && \
-		sudo ln -sF /Applications/Parity\ Ethereum.app/Contents/MacOS/parity /usr/local/bin/parity
+	  sudo cp $TMPDIR/parity /usr/local/bin && sudo chmod +x /usr/local/bin/parity
 	fi
```

Comment:
if PKG is neither of both it won't be installed without notification (e.g. capital letters)